infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -238,6 +245,29 @@
         this.initializationTaskThread.start();
     }
 
+    /**
+     * Triggers a new task to bootstrap the sequencer for the specified layout. If there is already
+     * a task in progress, this is a no-op.
+     *
+     * @param layout Layout to use to bootstrap the primary sequencer.
+     * @return Completable future which completes when the task completes successfully or with a
+     * failure.
+     */
+    private CompletableFuture<Boolean> triggerSequencerBootstrap(@NonNull Layout layout) {
```

Comment:
Should this be thread-safe? two threads can evaluate to sequencerRecoveryFuture.isDone()

Reply:
This is run in a single threaded manner. So we should not worry about thread safety. I will need to update it to an AtomicReference otherwise.

Reply:
Ok. The logic is async, you can just use synchronized on the method.